app/services/search_indexer.rb

```diff
@@ -14,63 +14,109 @@ def self.scrub_html_for_search(html)
     HtmlScrubber.scrub(html)
   end
 
-  def self.update_index(table, id, raw_data)
-    raw_data = Search.prepare_data(raw_data, :index)
+  # @param raw_entries The list of entries to index. When more than one entry is provided, they will get a weight. The
+  # first one will have a priority A; he second one a priority B, etc.
+  # When only 1 entry is provided, no weights will be used.
+  def self.update_index(table, id, *raw_entries)
+    raw_data = Search.prepare_data(raw_entries.join(' '), :index)
```

Comment:
This methods now supports providing multiple entries to index and, when more than 1 entry is provided, it will assign them a weight.

I didn't add support for defining the specific weights and, instead, used the convention of deducing it from the order. So the first entry gets an `A`. The second a `B`. And so. I think this works well for what we need and keeps the code simpler.